TACC/Frontera-Portal
# Task/FP-517 refactor exception handling in base view

## Overview: ##
Refactor exception handling in base view:
 * as it had a very broad exception clause that caused 500 to be returned on most errors
 * unique handling of ConnectionError/ConnectionError wasn't properly checking for a response
 
## PR Status: ##


## Related Jira tickets: ##

* [FP-517](https://jira.tacc.utexas.edu/browse/FP-517)

## Summary of Changes: ##
* Re-raise PermissionDenied and Http404 errors to allow django to handle
those errors instead of being squashed as 500s
* Fix handling of ConnectionError/ConnectionError to properly check for
response
* Add testing for base view
* Update unit tests for these status code changes 
* Update unit tests to use client

## Testing Steps: ##
1. Run tests and check coverage
2. Consider if these changes will have any unintended effects

## TODO ##
- [x] test uses of status-code in front end for `DataFilesTable/DataFilesTable.js` and consider if any extra unit tests are needed in back end
- [x] add extra to ApiException handling (see comments)
- [x] clean up calling of logger use (see comments)
